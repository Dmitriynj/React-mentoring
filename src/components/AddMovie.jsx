import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Input, InputLabel, Button } from '@material-ui/core';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },
  confirmButton: {
    marginLeft: 10,
    background: '#f65261',
    color: 'white',
    borderRadius: 4,
    border: '2px solid #f65261',
    '&:hover': { background: '#f65261' },
  },
  resetButton: {
    borderRadius: 4,
    border: '2px solid #f65261',
    color: '#f65261',
  },
}));

const AddMovie = () => {
  const classes = useStyles();

  return (
    <>
      <section className={classes.field}>
        <InputLabel className={classes.label}>MOVIE ID</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>TITLE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>RELEASE DATE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>MOVIE URL</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>GENRE</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel className={classes.label}>OVERVIEW</InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.field}>
        <InputLabel color="secondary" className={classes.label}>
          RUNTIME
        </InputLabel>
        <Input type="text" className={classes.input} />
      </section>
      <section className={classes.footer}>
        <Button className={classes.resetButton}>Reset</Button>
        <Button className={classes.confirmButton}>Submit</Button>
      </section>
    </>
  );
};

export { AddMovie };
