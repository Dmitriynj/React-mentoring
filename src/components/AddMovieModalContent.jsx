import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Input, InputLabel } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { createMovie } from '../store/thunks';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },

  ...MODAL_BUTTONS_STYLES,
}));

const AddNewMovieStateless = ({ sendCreateReq, onConfirm }) => {
  const classes = useStyles();
  const [creatingMovie, setCreatingMovie] = useState({
    title: '',
    release_date: '',
    overview: '',
    genres: [''],
    runtime: '',
    poster_path: '',
  });

  console.log('creating movie', creatingMovie);

  const onChangeInput = (event) => {
    const { name } = event.target;
    let { value } = event.target;
    if (name === 'runtime') {
      value = Number(value);
    }
    if (name === 'genres') {
      value = value.split(',');
    }
    setCreatingMovie({ ...creatingMovie, [name]: value });
  };

  const onSubmitChanges = () => {
    sendCreateReq(creatingMovie);
    onConfirm();
  };

  return (
    <>
      <section className={classes.field} key="title-field">
        <InputLabel className={classes.label}>TITLE</InputLabel>
        <Input
          name="title"
          type="text"
          onChange={onChangeInput}
          value={creatingMovie.title}
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="release-date-field">
        <InputLabel className={classes.label}>RELEASE DATE</InputLabel>
        <Input
          name="release_date"
          onChange={onChangeInput}
          value={creatingMovie.release_date}
          type="date"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="genres-field">
        <InputLabel className={classes.label}>GENRES</InputLabel>
        <Input
          name="genres"
          onChange={onChangeInput}
          value={creatingMovie.genres.join(',')}
          type="text"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="overview-field">
        <InputLabel className={classes.label}>OVERVIEW</InputLabel>
        <Input
          name="overview"
          onChange={onChangeInput}
          value={creatingMovie.overview}
          type="text"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="runtime-field">
        <InputLabel className={classes.label}>RUNTIME</InputLabel>
        <Input
          name="runtime"
          onChange={onChangeInput}
          value={creatingMovie.runtime}
          type="number"
          className={classes.input}
        />
      </section>
      <section className={classes.field} key="poster-path-field">
        <InputLabel className={classes.label}>POSTER PATH</InputLabel>
        <Input
          name="poster_path"
          onChange={onChangeInput}
          value={creatingMovie.poster_path}
          type="text"
          className={classes.input}
        />
      </section>

      <section className={classes.footer} key="footer">
        <Button onClick={onSubmitChanges} className={classes.confirmButton}>
          Create
        </Button>
      </section>
    </>
  );
};
AddNewMovieStateless.propTypes = {
  sendCreateReq: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  sendCreateReq: (data) => dispatch(createMovie(data)),
});

const AddMovieModalContent = connect(null, mapDispatchToProps)(AddNewMovieStateless);

export { AddMovieModalContent };
