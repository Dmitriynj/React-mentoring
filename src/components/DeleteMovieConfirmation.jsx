import React from 'react';
import { Typography, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  footer: { display: 'flex', justifyContent: 'flex-end', padding: 10 },

  ...MODAL_BUTTONS_STYLES,
}));

const DeleteMovieConfirmation = ({ onConfirm }) => {
  const classes = useStyles();

  return (
    <>
      <Typography>Are you really want to delete this movie?</Typography>
      <section className={classes.footer}>
        <Button onClick={onConfirm} className={classes.confirmButton}>
          Submit
        </Button>
      </section>
    </>
  );
};

DeleteMovieConfirmation.propTypes = {
  onConfirm: PropTypes.func,
};
DeleteMovieConfirmation.defaultProps = {
  onConfirm: () => {},
};

export { DeleteMovieConfirmation };
