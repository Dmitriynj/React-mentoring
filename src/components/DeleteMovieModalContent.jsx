import React from 'react';
import { Typography, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  footer: { display: 'flex', justifyContent: 'flex-end', padding: 10 },
  ...MODAL_BUTTONS_STYLES,
}));

const DeleteMovieModalContent = ({ onConfirm }) => {
  const classes = useStyles();

  return (
    <>
      <Typography>Are you really want to delete this movie?</Typography>
      <section className={classes.footer}>
        <Button onClick={onConfirm} className={classes.confirmButton}>
          Submit
        </Button>
      </section>
    </>
  );
};

DeleteMovieModalContent.propTypes = {
  onConfirm: PropTypes.func,
};
DeleteMovieModalContent.defaultProps = {
  onConfirm: () => {},
};

export { DeleteMovieModalContent };
