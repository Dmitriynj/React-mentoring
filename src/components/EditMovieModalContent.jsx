import React from 'react';
import { Button, Input, InputLabel } from '@material-ui/core';
import { uniqueId, map } from 'lodash';
import { makeStyles } from '@material-ui/core/styles';
import { MODAL_BUTTONS_STYLES } from '../constants';

const useStyles = makeStyles(() => ({
  field: {
    padding: '5px 0 5px 0',
  },
  input: {
    background: '#2e2e2e',
    color: 'white',
    fontWeight: 400,
    padding: '0 5px 0 5px',
    width: '100%',
    '&::after': {
      borderBottomColor: '#f65261',
    },
  },
  label: {
    fontWeight: 400,
    color: '#f65261',
    fontSize: 14,
    marginBottom: 2,
  },
  footer: {
    marginLeft: 'auto',
  },

  ...MODAL_BUTTONS_STYLES,
}));

const movieFields = [
  { name: 'TITLE' },
  { name: 'RELEASE DATE' },
  { name: 'MOVIE URL' },
  { name: 'GENRE' },
  { name: 'OVERVIEW' },
  { name: 'RUNTIME' },
];

const EditMovieModalContent = () => {
  const classes = useStyles();

  const fieldElement = ({ value, name }) => {
    return (
      <section className={classes.field} key={`field-${uniqueId()}`}>
        <InputLabel className={classes.label}>{name}</InputLabel>
        <Input defaultValue={value} type="text" className={classes.input} />
      </section>
    );
  };

  const fieldElements = map(movieFields, fieldElement);

  return (
    <>
      {fieldElements}
      <section className={classes.footer}>
        <Button className={classes.resetButton}>Reset</Button>
        <Button className={classes.confirmButton}>Submit</Button>
      </section>
    </>
  );
};

export { EditMovieModalContent };
