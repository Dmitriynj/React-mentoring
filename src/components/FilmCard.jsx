import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Card, CardMedia, CardHeader, Typography, CardContent } from '@material-ui/core';
import PropTypes from 'prop-types';
import { ManagedMovie } from './ManagedMovie';
import { useAppState } from '../hooks/useAppState';

const useStyles = makeStyles({
  card: {
    maxWidth: 220,
    backgroundColor: '#232323',
    color: '#eeeeee',
    margin: '5px auto',
  },
  cardHeaderTitle: {
    fontSize: 18,
  },
  wrapper: {
    position: 'relative',
  },
  moreContainer: {
    position: 'absolute',
    width: '100%',
    display: 'flex',
    justifyContent: 'flex-end',
  },
  more: {
    margin: 10,
    position: 'relative',
    backgroundColor: '#232323',
    borderRadius: '50%',
    color: 'white',
    height: 25,
    width: 25,
    '&:hover': {
      backgroundColor: '#232323',
    },
  },
  media: {
    cursor: 'pointer',
    height: 240,
  },
  actionButton: {
    color: 'white',
  },
  menuItem: {
    background: '#232323',
    color: 'white',
    '&:hover': {
      background: '#232323',
    },
  },
  description: {
    overflow: 'hidden',
    display: '-webkit-box',
    '-webkit-line-clamp': 3,
    '-webkit-box-orient': 'vertical',
  },
});

const FilmCard = ({ id, title, imageUrl, description, genres, releaseDate, avgVote, budget }) => {
  const classes = useStyles();
  const { setCurrentMovie } = useAppState();

  const openMovieDetails = () => {
    document.querySelector('#header-content').scrollIntoView({
      behavior: 'smooth',
    });
    setCurrentMovie({ title, imageUrl, description });
  };

  console.log('movie id', id);

  return (
    <Card className={classes.card}>
      <CardHeader
        classes={{ title: classes.cardHeaderTitle }}
        action={<ManagedMovie id={id} />}
        title={title}
      />
      <CardMedia
        className={classes.media}
        image={imageUrl}
        title="Contemplative Reptile"
        onClick={openMovieDetails}
      />
      <CardContent>
        <div className={classes.description}>
          <Typography variant="body2" component="p">
            {description}
          </Typography>
        </div>
        <Typography variant="body2" component="p">
          Genres: {genres.join(',')}
        </Typography>
        <Typography variant="body2" component="p">
          Release date: {releaseDate}
        </Typography>
        <Typography variant="body2" component="p">
          Avg vote: {avgVote}
        </Typography>
        <Typography variant="body2" component="p">
          Budget: {budget}
        </Typography>
      </CardContent>
    </Card>
  );
};

FilmCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  genres: PropTypes.array.isRequired,
  releaseDate: PropTypes.string.isRequired,
  avgVote: PropTypes.number,
  budget: PropTypes.number,
};
FilmCard.defaultProps = {
  imageUrl: '',
  avgVote: 0,
  budget: 0,
};

export { FilmCard };
