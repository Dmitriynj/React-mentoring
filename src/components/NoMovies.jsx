import React from 'react';
import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  template: {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 1 auto',
    justifyContent: 'center',
    alignItems: 'center',
    color: 'white',
  },
}));

const NoMovies = () => {
  const classes = useStyles();

  return (
    <div className={classes.template}>
      <Typography variant="h4">No movies found</Typography>
    </div>
  );
};

export { NoMovies };
