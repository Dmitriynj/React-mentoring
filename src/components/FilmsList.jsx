import React from 'react';
import Grid from '@material-ui/core/Grid';
import { reduce, uniqueId, isEmpty } from 'lodash';
import { FilmCard } from './FilmCard';

const mockedFilmsList = [
  {
    title: 'Some',
    description: 'Description',
    imageUrl: 'sample.jpg',
  },
  {
    title: 'Some1',
    description: 'Description1',
    imageUrl: 'sample1.jpg',
  },
  {
    title: 'Some1',
    description: 'Description2',
    imageUrl: 'sample2.jpg',
  },
  {
    title: 'Some3',
    description: 'Description3',
    imageUrl: 'sample3.jpg',
  },
  {
    title: 'Some4',
    description: 'Description4',
    imageUrl: 'sample4.jpg',
  },
  {
    title: 'Some5',
    description: 'Description5',
    imageUrl: 'sample5.jpg',
  },
  {
    title: 'Some6',
    description: 'Description6',
    imageUrl: 'sample6.jpg',
  },
  {
    title: 'Some7',
    description: 'Description7',
    imageUrl: 'sample7.jpg',
  },
  {
    title: 'Some8',
    description: 'Description8',
    imageUrl: 'sample8.jpg',
  },
  {
    title: 'Some9',
    description: 'Description8',
    imageUrl: 'sample9.jpg',
  },
  {
    title: 'Some10',
    description: 'Description8',
    imageUrl: 'sample10.jpg',
  },
  {
    title: 'Some11',
    description: 'Description8',
    imageUrl: 'sample11.jpg',
  },
  {
    title: 'Some12',
    description: 'Description8',
    imageUrl: 'sample12.jpg',
  },
];

const getFilmsPortion = (filmPortion) => {
  return filmPortion.map(({ title, description, imageUrl }) => (
    <Grid item xs={3} key={`film-row-item-${uniqueId()}`}>
      <FilmCard title={title} description={description} imageUrl={imageUrl} />
    </Grid>
  ));
};

const filmElements = reduce(
  mockedFilmsList,
  ({ result, temp }, filmData, index) => {
    if (index % 4 === 0 && !isEmpty(temp)) {
      return {
        result: [
          ...result,
          <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
            {getFilmsPortion(temp)}
          </Grid>,
        ],
        temp: [filmData],
      };
    }

    if (index === mockedFilmsList.length - 1) {
      return {
        result: [
          ...result,
          <Grid container item xs={12} key={`films-row-${uniqueId()}`}>
            {getFilmsPortion([...temp, filmData])}
          </Grid>,
        ],
        temp: [],
      };
    }

    return {
      result,
      temp: [...temp, filmData],
    };
  },
  { result: [], temp: [] }
);

const FilmsList = () => {
  return <Grid container>{filmElements.result}</Grid>;
};

export { FilmsList };
