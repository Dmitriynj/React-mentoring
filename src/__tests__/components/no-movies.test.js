import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render } from '../test-utils';
import { NoMovies } from '../../components/NoMovies';

function renderComponent() {
  return render(
    <MemoryRouter initialEntries={['/movies']}>
      <Route exact path="/movies">
        <NoMovies />
      </Route>
    </MemoryRouter>,
    { initialState: {} }
  );
}

describe('NoMovies', () => {
  it('renders correctly', () => {
    const { container, queryByText } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
    expect(queryByText('No movies found')).toMatchSnapshot();
  });
});
