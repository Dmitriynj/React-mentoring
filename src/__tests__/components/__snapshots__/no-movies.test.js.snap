// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMovies renders correctly 1`] = `
<div
  class="makeStyles-template-1"
>
  <h4
    class="MuiTypography-root MuiTypography-h4"
  >
    No movies found
  </h4>
</div>
`;

exports[`NoMovies renders correctly 2`] = `
<h4
  class="MuiTypography-root MuiTypography-h4"
>
  No movies found
</h4>
`;
